File: Exam-Service/src/main/java/com/exam_service/Exam_Service/service/ExamService.java
package com.exam_service.Exam_Service.service;

import com.exam_service.Exam_Service.dto.internal.ExamAnswerKeyResponse;
import com.exam_service.Exam_Service.dto.request.CreateExamRequest;
import com.exam_service.Exam_Service.dto.request.CreateQuestionRequest;
import com.exam_service.Exam_Service.dto.response.*;
import org.springframework.data.domain.Page;

public interface ExamService {

    CreateExamResponse createExam(CreateExamRequest request);
    QuestionResponse addQuestion(Long examId, CreateQuestionRequest request);

    ExamResponse getExamById(Long examId);

    Page<ExamSummaryResponse> getAllExams(
            int page,
            int size
    );

   PublishedExamResponse publishedExam(Long examId);

   public void deleteExamById(Long examId);

   ExamAnswerKeyResponse getExamAnswerKey(Long examId);
}

File: frontend/src/services/examService.js
import api from "../api/axios";

export const getAllExams = async (page = 0, size = 10) => {
  const response = await api.get("/api/exams", {
    params: {
      page,
      size,
    },
  });

  return response.data;
};

export const createExam = async (examData) => {
  const response = await api.post("/api/exams", examData);

  return response.data;
};

export const getExamById = async (id) => {
  const response = await api.get(`/api/exams/${id}`);

  return response.data;
};

export const deleteExam = async (examId) => {
  await api.delete(`/api/exams/${examId}`);
};
